File: repos_estudiantes_ALVAREZ_primera_entrega/ARUQUIPA_VARGAS_ANA_GLORIA/HERENCIA/Ejercicio3py_java/Animal.java
package Ejercicio3py_java;

public class Animal {
    protected String nombre;
    protected int edad;

    public Animal(String nombre, int edad) {
        this.nombre = nombre;
        this.edad = edad;
    }

    public void desplazarse() {
        System.out.println(nombre + " se desplaza de una manera genérica.");
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/ARUQUIPA_VARGAS_ANA_GLORIA/HERENCIA/Ejercicio3py_java/Animal.py
class Animal:
    def __init__(self, nombre, edad):
        self.nombre = nombre
        self.edad = edad

    def desplazarse(self):
        print(f"{self.nombre} se desplaza")

class Leon(Animal):
    def desplazarse(self):
        print(f"{self.nombre} camina")

class Pinguino(Animal):
    def desplazarse(self):
        print(f"{self.nombre} nada")

class Canguro(Animal):
    def desplazarse(self):
        print(f"{self.nombre} salta")

animales = [
    Leon("Simba", 5),
    Pinguino("Magu", 3),
    Canguro("Kanga", 4)
]

for animal in animales:
    animal.desplazarse()
